[BUG] cannot install mkl (doc dependency) on Mac M1 Sillicon chip

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Operating system

- [ ] Linux
- [X] Mac
- [ ] Windows

### Operating system version

For example one of the following:
- Linux Ubuntu 22.04
- Mac OS Version 12 "monterey"
- Windows 11


### Python version

- [ ] 3.11
- [ ] 3.10
- [X] 3.9
- [ ] 3.8
- [ ] 3.7

### nilearn version

0.10.2dev

commit: ef66712544ef3ff99be88aa3958b53e0ff14c797

### Expected behavior

pip install .[doc] to work

### Current behavior & error messages

fails because mkl dependency seems unavailable on Mac M1

### Steps and code to reproduce bug

_No response_

Some test are failing possibly relate to M1 as well:

================================================ short test summary info =================================================
FAILED nilearn/_utils/tests/test_testing.py::test_memory_usage - AssertionError: Regex pattern did not match.
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_standardize[False-True-zscore] - AssertionError: assert 0.4175871156919765 < 0.2
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_standardize[False-True-psc] - AssertionError: assert 0.32822545500593064 < 0.2
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_standardize[True-True-zscore] - AssertionError: assert 0.27566230766502103 < 0.2
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_standardize[True-True-psc] - AssertionError: assert 0.2624108924225602 < 0.2
==================== 5 failed, 2626 passed, 5 skipped, 2 xfailed, 1457 warnings in 847.89s (0:14:07) =====================

Having the same issue, hardware info: 
<img width="220" alt="image" src="https://github.com/nilearn/nilearn/assets/78616918/dc1e33b8-bb35-4af2-8ae4-ee935ac5652b">


Error here, full error message below:

`ValueError: Memory profiler measured an untrustable memory consumption (0.00 MiB). The expected memory limit was 100.00 MiB. Try to bench with larger objects (at least 100MiB in memory).

nilearn/_utils/testing.py:104: ValueError
`


```
================== FAILURES ==================
_____________ test_memory_usage ______________

    @with_memory_profiler
    def test_memory_usage():
        # Valid measures (larger objects)
        for mem in (500, 200):
            assert_memory_less_than(mem, 0.1, create_object, mem * 1024**2)
    
        # Ensure an exception is raised with too small objects as
        # memory_profiler can return non trustable memory measure in this case.
        with pytest.raises(
            ValueError, match="Memory profiler measured an untrustable memory"
        ):
            assert_memory_less_than(50, 0.1, create_object, 25 * 1024**2)
    
        # Ensure ValueError is raised if memory used is above expected memory
        # limit.
        with pytest.raises(ValueError, match="Memory consumption measured"):
>           assert_memory_less_than(100, 0.1, create_object, 200 * 1024**2)

nilearn/_utils/tests/test_testing.py:33: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

memory_limit = 100, tolerance = 0.1
callable_obj = <function create_object at 0x12cc43ce0>
args = (209715200,), kwargs = {}
mem_used = 0.0

    def assert_memory_less_than(
        memory_limit, tolerance, callable_obj, *args, **kwargs
    ):
        """Check memory consumption of a callable stays below a given limit.
    
        Parameters
        ----------
        memory_limit : int
            The expected memory limit in MiB.
    
        tolerance : float
            As memory_profiler results have some variability, this adds some
            tolerance around memory_limit. Accepted values are in range [0.0, 1.0].
    
        callable_obj : callable
            The function to be called to check memory consumption.
    
        """
        mem_used = memory_used(callable_obj, *args, **kwargs)
    
        if mem_used > memory_limit * (1 + tolerance):
            raise ValueError(
                f"Memory consumption measured ({mem_used:.2f} MiB) is "
                f"greater than required memory limit ({memory_limit} MiB) within "
                f"accepted tolerance ({tolerance * 100:.2f}%)."
            )
    
        # We are confident in memory_profiler measures above 100MiB.
        # We raise an error if the measure is below the limit of 50MiB to avoid
        # false positive.
        if mem_used < 50:
>           raise ValueError(
                "Memory profiler measured an untrustable memory "
                f"consumption ({mem_used:.2f} MiB). The expected memory "
                f"limit was {memory_limit:.2f} MiB. Try to bench with larger "
                "objects (at least 100MiB in memory)."
            )
E           ValueError: Memory profiler measured an untrustable memory consumption (0.00 MiB). The expected memory limit was 100.00 MiB. Try to bench with larger objects (at least 100MiB in memory).

nilearn/_utils/testing.py:104: ValueError

During handling of the above exception, another exception occurred:

    @with_memory_profiler
    def test_memory_usage():
        # Valid measures (larger objects)
        for mem in (500, 200):
            assert_memory_less_than(mem, 0.1, create_object, mem * 1024**2)
    
        # Ensure an exception is raised with too small objects as
        # memory_profiler can return non trustable memory measure in this case.
        with pytest.raises(
            ValueError, match="Memory profiler measured an untrustable memory"
        ):
            assert_memory_less_than(50, 0.1, create_object, 25 * 1024**2)
    
        # Ensure ValueError is raised if memory used is above expected memory
        # limit.
>       with pytest.raises(ValueError, match="Memory consumption measured"):
E       AssertionError: Regex pattern did not match.
E        Regex: 'Memory consumption measured'
E        Input: 'Memory profiler measured an untrustable memory consumption (0.00 MiB). The expected memory limit was 100.00 MiB. Try to bench with larger objects (at least 100MiB in memory).'

nilearn/_utils/tests/test_testing.py:32: AssertionError
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -63,7 +63,6 @@ doc = [
     "flake8-docstrings",
     "furo",
     "memory_profiler",  # measuring memory during docs building
-    "mkl",
     "myst-parser",
     "numpydoc",
     "ruamel.yaml",

